Samples/dotnet.eShop/DotNetEShop.Cli/Properties/launchSettings.json

```diff
@@ -1,8 +1,16 @@
 {
   "profiles": {
-    "dotnet.eShop.Architecture.Cli": {
-      "commandName": "Project",
-      "commandLineArgs": "execute-aac-strategy --aac-root-namespace \"dotnet.eShop.Architecture\" --aac-output-path \"C:\\C4InterFlow\\Samples\\dotnet.eShop\\dotnet.eShop.Architecture\\Yaml\" --aac-writer-strategy \"dotnet.eShop.Architecture.Cli.CSharpToYamlBasketApiAaCGenerator, dotnet.eShop.Architecture.Cli\" --params software-system-source-path=\"C:\\Data\\Projects\\C4InterFlow\\eShop-main\\src\\Basket.API\\Basket.API.csproj\" --params software-system-name=\"BasketApi\""
+    "DotNetEShop - Execute AaC Strategy": {
```

Comment:
I'm getting the following error:

[14:00:23 ERR] execute-aac-strategy command execution failed: One or more errors occurred. (An exception of type System.InvalidOperationException was thrown: Failed to find all versions of .NET Core MSBuild. Call to hostfxr_resolve_sdk2. There may be more details in stderr.)
System.AggregateException: One or more errors occurred. (An exception of type System.InvalidOperationException was thrown: Failed to find all versions of .NET Core MSBuild. Call to hostfxr_resolve_sdk2. There may be more details in stderr.) ---> Microsoft.CodeAnalysis.MSBuild.Rpc.RemoteInvocationException: An exception of type System.InvalidOperationException was thrown: Failed to find all versions of .NET Core MSBuild. Call to hostfxr_resolve_sdk2. There may be more details in stderr.
   at Microsoft.CodeAnalysis.MSBuild.Rpc.RpcClient.InvokeCoreAsync(Int32 targetObject, String methodName, List`1 parameters, Type expectedReturnType, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.MSBuild.Rpc.RpcClient.InvokeAsync[T](Int32 targetObject, String methodName, List`1 parameters, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.MSBuild.Rpc.RemoteBuildHost.LoadProjectFileAsync(String projectFilePath, String languageName, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.DoOperationAndReportProgressAsync[TResult](ProjectLoadOperation operation, String projectPath, String targetFramework, Func`1 doFunc)
   at Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.LoadProjectFileInfosAsync(String projectPath, DiagnosticReportingOptions reportingOptions, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.LoadProjectInfosFromPathAsync(String projectPath, DiagnosticReportingOptions reportingOptions, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.LoadAsync(CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.LoadProjectInfoAsync(String projectFilePath, ProjectMap projectMap, IProgress`1 progress, ILogger msbuildLogger, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.LoadProjectInfoAsync(String projectFilePath, ProjectMap projectMap, IProgress`1 progress, ILogger msbuildLogger, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.OpenProjectAsync(String projectFilePath, ILogger msbuildLogger, IProgress`1 progress, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at C4InterFlow.Automation.Writers.CSharpToAnyAaCWriter..ctor(String softwareSystemSourcePath, String architectureRootNamespace) in C:\C4InterFlow-kopelli\C4InterFlow.Automation\Writers\CSharpToAnyAaCWriter.cs:line 54
   at C4InterFlow.Automation.Writers.CSharpToYamlAaCWriter..ctor(String softwareSystemSolutionPath, String architectureRootNamespace, String architectureOutputPath) in C:\C4InterFlow-kopelli\C4InterFlow.Automation\Writers\CSharpToYamlAaCWriter.cs:line 18
   at DotNetEShop.Cli.CSharpToYamlBasketApiAaCGenerator.Execute() in C:\C4InterFlow-kopelli\Samples\dotnet.eShop\DotNetEShop.Cli\CSharpToYamlBasketApiAaCGenerator.cs:line 18
   at C4InterFlow.Automation.AaCWriterContext.ExecuteStrategy() in C:\C4InterFlow-kopelli\C4InterFlow\Automation\AaCWriterContext.cs:line 30
   at C4InterFlow.Cli.Commands.ExecuteAaCStrategyCommand.Execute(String architectureRootNamespace, String architectureOutputPath, String architectureAsCodeWriterStrategyType, Dictionary`2 architectureAsCodeParams) in C:\C4InterFlow-kopelli\C4InterFlow\Cli\Commands\ExecuteAaCStrategyCommand.cs:line 59

Reply:
What's interesting is when I run it, I get the following (with a few uncaught exceptions raised in the `InternalLoad` method within System.Relection.RuntimeAssembly):

```
[08:43:22 INF] execute-aac-strategy command is executing
[08:43:24 INF] Loading Software System source from C:\src\_scratch\eShop\src\Basket.API\Basket.API.csproj
[08:43:24 WRN] MSBuildLocator could not find any instances to register.
[08:44:22 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi.yaml already exists
[08:44:22 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Grpc.yaml already exists
Adding Grpc Container, Component and Interface objects for 'Basket.API' project...
[08:44:23 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Grpc\Components\BasketService.yaml already exists
[08:44:39 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Grpc\Components\BasketService\Interfaces\GetBasket.yaml already exists
[08:44:39 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Grpc\Components\BasketService\Interfaces\UpdateBasket.yaml already exists
[08:44:39 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Grpc\Components\BasketService\Interfaces\DeleteBasket.yaml already exists
[08:44:40 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Grpc\Components\BasketService\Interfaces\ThrowNotAuthenticated.yaml already exists
[08:44:40 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Grpc\Components\BasketService\Interfaces\ThrowBasketDoesNotExist.yaml already exists
[08:44:40 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Grpc\Components\BasketService\Interfaces\MapToCustomerBasketResponse.yaml already exists
[08:44:40 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Grpc\Components\BasketService\Interfaces\MapToCustomerBasket.yaml already exists
[08:44:40 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Data.yaml already exists
Adding Data Container, Component and Interface objects for 'Basket.API' project...
[08:44:40 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Data\Components\RedisBasketRepository.yaml already exists
[08:44:40 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Data\Components\RedisBasketRepository\Interfaces\GetBasketKey.yaml already exists
[08:44:40 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Data\Components\RedisBasketRepository\Interfaces\DeleteBasketAsync.yaml already exists
[08:44:40 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Data\Components\RedisBasketRepository\Interfaces\GetBasketAsync.yaml already exists
[08:44:40 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Data\Components\RedisBasketRepository\Interfaces\UpdateBasketAsync.yaml already exists
[08:44:40 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Data\Components\RedisDatabase.yaml already exists
[08:44:40 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Data\Components\RedisDatabase\Interfaces\KeyDeleteAsync.yaml already exists
[08:44:40 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Data\Components\RedisDatabase\Interfaces\StringGetLeaseAsync.yaml already exists
[08:44:40 WRN] Document C:\src\kopelli\C4InterFlow\Samples\dotnet.eShop\DotNetEShop.Cli\\..\DotNetEShop\Yaml\SoftwareSystems\BasketApi\Containers\Data\Components\RedisDatabase\Interfaces\StringSetAsync.yaml already exists
Adding Software System Type Mappings...
Updating Flow property in all Interface objects...
[08:44:40 INF] Strategy execution completed
```

The assemblies seem to be 

* 'Could not load file or assembly 'Microsoft.CodeAnalysis.VisualBasic.Workspaces, Version=4.9.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.'
* 'Could not load file or assembly 'Microsoft.CodeAnalysis.Features, Version=4.9.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.'
* 'Could not load file or assembly 'Microsoft.CodeAnalysis.CSharp.Features, Version=4.9.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.'
* 'Could not load file or assembly 'Microsoft.CodeAnalysis.VisualBasic.Features, Version=4.9.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.'
* Could not load file or assembly 'Microsoft.Interop.ComInterfaceGenerator, Version=8.0.10.36612, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL'. The system cannot find the file specified.'
* 'Could not load file or assembly 'Microsoft.Interop.SourceGeneration, Version=8.0.10.36612, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.'
* 'Could not load file or assembly 'Microsoft.Interop.JavaScript.JSImportGenerator, Version=8.0.10.36612, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL'. The system cannot find the file specified.'
* 'Could not load file or assembly 'Microsoft.Interop.LibraryImportGenerator, Version=8.0.10.36612, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL'. The system cannot find the file specified.'
* 'Could not load file or assembly 'System.Text.Json.SourceGeneration, Version=8.0.10.36612, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL'. The system cannot find the file specified.'
* 'Could not load file or assembly 'System.Text.RegularExpressions.Generator, Version=8.0.10.36612, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL'. The system cannot find the file specified.'
* 'Could not load file or assembly 'Microsoft.Extensions.Logging.Generators, Version=8.0.10.36612, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL'. The system cannot find the file specified.'
* 'Could not load file or assembly 'Microsoft.Extensions.Options.SourceGeneration, Version=8.0.10.36612, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL'. The system cannot find the file specified.'
* 'Could not load file or assembly 'Microsoft.CodeAnalysis.Razor.Compiler, Version=9.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL'. The system cannot find the file specified.'


Within VS, I do have "Enable Just My Code" unchecked thus why I'm seeing those missing FileNotFoundExceptions.

Reply:
Resolved via https://github.com/SlavaVedernikov/C4InterFlow/pull/104#discussion_r1778683033